spec/tapioca/compilers/dsl/active_record_scope_spec.rb

```diff
@@ -0,0 +1,104 @@
+# typed: false
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe("Tapioca::Compilers::Dsl::ActiveRecordScope") do
+  before(:each) do
+    require "tapioca/compilers/dsl/active_record_scope"
+  end
+
+  subject do
+    Tapioca::Compilers::Dsl::ActiveRecordScope.new
+  end
+
+  describe("#initialize") do
+    def constants_from(content)
+      with_content(content) do
+        subject.processable_constants.map(&:to_s).sort
+      end
+    end
+
+    it("gathers no constants if there are no ActiveRecord classes") do
+      assert_empty(subject.processable_constants)
+    end
+
+    it("gathers only ActiveRecord constants with no abstract classes") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+        end
+
+        class Product < ActiveRecord::Base
+          self.abstract_class = true
+        end
+
+        class User
+        end
+      RUBY
+
+      assert_equal(["Post"], constants_from(content))
+    end
+  end
+
+  describe("#decorate") do
+    def rbi_for(content)
+      with_content(content) do
+        parlour = Parlour::RbiGenerator.new(sort_namespaces: true)
+        subject.decorate(parlour.root, Post)
+        parlour.rbi
+      end
+    end
+
+    it("generates RBI file for ActiveRecord classes with a scope field") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          #extend Post::GeneratedRelationMethods
```

Comment:
We don't want these `extend`s in the source file. We want to get the `extend` line in the RBI output.